bundles/test/org.eclipse.smarthome.magic/src/main/java/org/eclipse/smarthome/magic/binding/internal/firmware/MagicFirmwareProvider.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.magic.binding.internal.firmware;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.firmware.Firmware;
+import org.eclipse.smarthome.core.thing.binding.firmware.FirmwareBuilder;
+import org.eclipse.smarthome.core.thing.firmware.FirmwareProvider;
+import org.eclipse.smarthome.magic.binding.MagicBindingConstants;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * Provides firmware for the magic thing type for firmware update.
+ *
+ * @author Dimitar Ivanov - Initial contribution
+ */
+@Component(service = FirmwareProvider.class)
+public class MagicFirmwareProvider implements FirmwareProvider {
+
+    private final Map<ThingTypeUID, Set<Firmware>> magicFirmwares = new HashMap<ThingTypeUID, Set<Firmware>>();
```

Comment:
Make it a set and think about initializing it statically

Reply:
See my comment above. If you think that we do not need extra extensions here - I will leave it as simple as possible.

Reply:
As @htreu is also with you on that I will make it as simple as possible.